2022.5.0 (#22619)
home-assistant/home-assistant.github.io: source/_posts/2022-05-04-release-20225.markdown

<!-- TODO REMOVE BELOW BEFORE RELEASE AND CHANGE TITLE/DATE IN FRONTMATTER -->

These are the beta release notes for Home Assistant Core 2022.5 (and is thus a
work in progress).

If you encounter any issues with the beta release, please report them on GitHub:

- Issues with integrations, automations and such (Core related):<br>
  <https://github.com/home-assistant/core/issues>
- Issues with the frontend/Lovelace:<br>
  <https://github.com/home-assistant/frontend/issues>
- Issues with the Supervisor:<br>
  <https://github.com/home-assistant/supervisor/issues>
- Issues with the documentation:<br>
  <https://github.com/home-assistant/home-assistant.io/issues>

Please be sure to include the beta version you are running in the issue
version field (not title), so we can classify your issue correctly.

Issues introduced in the beta are processed with priority.

---

<!-- TODO REMOVE ABOVE BEFORE RELEASE AND CHANGE TITLE/DATE IN FRONTMATTER -->

Home Assistant Core 2022.5!

And, as often said: "All things seem possible in May!". Well, possibilities
we have for you this last month of spring.

> The most visible thing this release is the next iteration of the setting menu,

Reviewer: ```suggestion
The most visible thing this release is the next iteration of the settings menu,
```

Author: Thanks!

<!-- TODO REMOVE BELOW BEFORE RELEASE AND CHANGE TITLE/DATE IN FRONTMATTER -->

These are the beta release notes for Home Assistant Core 2022.5 (and is thus a
work in progress).

If you encounter any issues with the beta release, please report them on GitHub:

- Issues with integrations, automations and such (Core related):<br>
  <https://github.com/home-assistant/core/issues>
- Issues with the frontend/Lovelace:<br>
  <https://github.com/home-assistant/frontend/issues>
- Issues with the Supervisor:<br>
  <https://github.com/home-assistant/supervisor/issues>
- Issues with the documentation:<br>
  <https://github.com/home-assistant/home-assistant.io/issues>

Please be sure to include the beta version you are running in the issue
version field (not title), so we can classify your issue correctly.

Issues introduced in the beta are processed with priority.

---

<!-- TODO REMOVE ABOVE BEFORE RELEASE AND CHANGE TITLE/DATE IN FRONTMATTER -->

Home Assistant Core 2022.5!

And, as often said: "All things seem possible in May!". Well, possibilities
we have for you this last month of spring.

The most visible thing this release is the next iteration of the settings menu,

> of which the result, to make personally, makes tons of sense. It took me a bit

Reviewer 2: ```suggestion
of which the result, to me personally, makes tons of sense. It took me a bit
```

- [Gauge card segment colors](#gauge-card-segment-colors)
- [Database optimizations](#database-optimizations)
- [Update entity updates](#update-entity-updates)
- [Insteon control panel](#insteon-control-panel)
- [Template selector](#template-selector)
- [Other noteworthy changes](#other-noteworthy-changes)
- [New Integrations](#new-integrations)
- [Integrations now available to set up from the UI](#integrations-now-available-to-set-up-from-the-ui)
- [Need help? Join the community!](#need-help-join-the-community)
- [Breaking Changes](#breaking-changes)
- [Farewell to the following](#farewell-to-the-following)
- [All changes](#all-changes)

## Reorganized settings menu

A couple of releases ago, we started reorganizing the configuration menu, and
this release is bringing the next big step in that reorganization.

The goal is to have everything in a single logical place, and after user tests
and many discussions in frontend and UX channels, this is what the menu
now looks like:

<img class="no-shadow" src='/images/blog/2022-05/configuration-menu.png' alt='Screenshot showing the new configuration menu structure'>

> Most notably, it renamed from "Configuration" to "Settings", and we have a

Reviewer 2: ```suggestion
Most notably, it's been renamed from "Configuration" to "Settings", and we have a
```

### If-then

When we introduced the [Choose action](/docs/scripts/#choose-a-group-of-actions),
our goal was to provide a structure that allowed for other action sequences
selectively based on conditions.

While this structure is very flexible and extensive, there was still a desire
for an if-then(-else) structure that is small, simple, compact, and clean.
This release brings you just that.

The new if-then action is available via YAML and via the UI using automations
and scripts editors.

<img class="no-shadow" src='/images/blog/2022-05/if-then.png' alt='Screenshot showing If-then'>

{% details "If-then example in YAML" %}

If YAML automations are more your thing, this is how you can use the new
if-then action in your automations and scripts.

```yaml
actions:
  - if:
      - alias: "If no one is home"
        condition: state
        entity_id: zone.home
        state: 0
    then:
      - alias: "Then start cleaning already!"
        service: vacuum.start
        target:
          area_id: living_room
    # The `else` is fully optional and can be omitted
    else:
      - service: notify.notify
        data:
          message: "Skipped cleaning, someone is home!"
```

> Note that that `if` also supports a shorthand condition templates (if that

Reviewer 2: ```suggestion
Note that that `if` also supports a shorthand condition template (if that
```

    - service: notify.phone
      data:
        title: "Message in {{ repeat.item.language }}"
        message: "{{ repeat.item.message }}!"
```

{% endraw %}

Each item in the list will be run against a sequence of actions, and the item
is available as a variable you can use in your templates. The items you can
provide to `for_each` can be mappings, lists of just strings, and even complex
templates that provide a list as a result.

[More information can be found in the scripts documentation](/docs/scripts/#for-each).

[group of actions]: /docs/scripts/#repeat-a-group-of-actions

### Disable any trigger, condition, or action

Sometimes, you may want to disable a specific trigger, action, or condition,
whether this is for testing, a temporary workaround, or any other reason.

In YAML, you'd comment out parts of your automation, but if you wanted
to do that in the UI, the only option you have is to delete it from the
automation or script.

In this release, we added support for disabling a trigger, action, or condition;
without the need for removing it or commenting it out! A disabled trigger
won't fire, a disabled condition always passes, and a disabled action is
skipped.

> <img class="no-shadow" src='/images/blog/2022-05/disabled-condition.png' alt='Screenshot showing an disabled condition in an UI automation'>

Reviewer 2: ```suggestion
<img class="no-shadow" src='/images/blog/2022-05/disabled-condition.png' alt='Screenshot showing a disabled condition in a UI automation'>
```

{% details "Disabled example in YAML" %}

If YAML automations are more your thing, this disabled feature is still

> helpful. While, of course, you can still comment parts out easily; using the

Reviewer 2: ```suggestion
helpful. While, of course, you can still comment parts out easily; using
```

      - platform: state
        entity_id: binary_sensor.motion
        to: "on"
    action:
      - parallel:
          - service: notify.person1
            data:
              message: "These messages are sent at the same time!"
          - service: notify.person2
            data:
              message: "These messages are sent at the same time!"
```

This feature is partly available via the UI; however, as said: This is quite
a powerful and advanced feature, and it comes with caveats. Be sure to check
out the [script documentation on parallelizing actions](/docs/scripts/#parallelizing-actions)
before deciding to use it.

### Using a single state trigger for multiple entities

If you write automations in YAML, you are probably already aware of the
ability to trigger on multiple entities in a single trigger; it has been
around for quite some time.

And now also available in the UI. A small addition that might help you cut
down the length of your UI-managed automations.

<img class="no-shadow" src='/images/blog/2022-05/trigger-multiple-entities.png' alt='Screenshot showing multiple entites in a single trigger from the UI.'>

### Trigger on not matching to/from states

> This is an YAML only feature we have added to the state triggers: Triggering

Reviewer 2: ```suggestion
This is a YAML only feature we have added to the state triggers: Triggering
```

on not matching to/from states. Yes, you read that correctly. When it **not**
matches it triggers.

Instead of `from`, you can now use `not_from` and instead of `to`, you can now

> use `not_to`. This example trigger will only from if the state was previously

Reviewer 2: ```suggestion
use `not_to`. This example trigger will only trigger if the state was previously
```

not "unavailable" or "unknown":

```yaml
trigger:
  - platform: state
    entity_id: light.living_room
    not_from:
      - "unavailable"
      - "unknown"
    to: "on"
```

[More information can be found on the state condition documentation](/docs/scripts/conditions#state-condition).

### Shorthand notation for logical conditions

A neat little YAML feature [@thomasloven] added can make your YAML-based
automations and scripts looks significantly cleaner.

Logical conditions (also known as `and`, `or`, `not`) now have a shorthand
notation. Some example pseudo code to show them all:

```yaml
or:
  - <condition>
  - and:
    - <condition>
    - <condition>
  - not:
    - <condition>
```

In the above `<condition>`, of course, needs to be replaced with an actual

> condition, but the short new syntax of `or`, `and`, and `not` clearly visible.

Reviewer 2: ```suggestion
condition, but the short new syntax of `or`, `and`, and `not` is clearly visible.
```

Screenshot of the gauge card with multiple colored segments.
</p>

Helpful if you'd like to define upper and lower sensor
limits on your gauge, as shown in the example above. The configuration example
used for the above screenshot can be found in the [Gauge Card documentation].

[@Netzwerkfehler]: https://github.com/Netzwerkfehler
[Gauge Card documentation]: /dashboards/gauge/#examples

## Database optimizations

The last release focused on reducing the size of the database and optimizing
the writing of data to the database.

While there are some additional optimizations in this release to further reduce
the amount of data that needs to be written, in this release, we focused on how
often data is read from the database and optimizing its scale for larger setups.

This release is for you if you have many sensors generating statistics, as
compiling statistics now takes 30-100x less time.

Are you using the [History Stats] integration? The number of database queries
needed for most sensors with a fixed start time is 99% less.

Even if your instance is small and you don’t record many states, you’ll benefit

> from the recorder optimizations to avoid waking up when there is no new

Reviewer 2: What is waking up? A thread?

Author: Good question...

I think we can just delete this paragraph actually (story remains the same without it)

  - All logs moved to `Logs` (Supervisor, audio, etc).
  - `Backups` moved from the Supervisor/Main menu to here.
  - `Network` now has all network related settings previously in the General
    settings and Supervisor.
  - `Storage` provides information about used space, eMMC Lifetime, and also
    provides the `Move datadisk` feature in the overflow menu*.
  - `Hardware` now has the `Reboot` and `Restart Host` controls,
    `Available Hardware` is available in the overflow menu*.
  - `System Health` moved to its own menu item in the system menu.
  - Processor and memory usage was moved into the new `System Health` menu.
  - The integration list with timings, previously shown in the `About` menu, is
    moved into the new `System Health` menu.

Additionally, the developers tools has a brand new `YAML` tab, which contains
the buttons and tools to reload and check your YAML configuration. This was
previously known as "Server Controls", but now moved to the developer tools.

\* The overflow menu is the menu you see when you click on the three vertical
   dots in the top right of your screen.

{% enddetails %}

{% details "AVM FRITZ!Box Call Monitor" %}

> The previously deprecated YAML configuration of theAVM FRITZ!Box Call Monitor

Reviewer 2: ```suggestion
The previously deprecated YAML configuration of the AVM FRITZ!Box Call Monitor
```

[#67003]: https://github.com/home-assistant/core/pull/67003

{% enddetails %}

{% details "deCONZ" %}

Migrating deCONZ Siren entities from switch platform to siren platform has
been removed. This only affects users upgrading from before 2021.10 to 2022.5
or later. At most, this causes lingering unavailable switch entities; which
can be manually removed.

([@Kane610] - [#70600]) ([documentation](/integrations/deconz))

[@Kane610]: https://github.com/Kane610
[#70600]: https://github.com/home-assistant/core/pull/70600

---

Migrated deCONZ light entities of type "On/Off Output" to switch
from light platform as they are binary devices (only on/off).

([@Kane610] - [#70598]) ([documentation](/integrations/deconz))

[@Kane610]: https://github.com/Kane610
[#70598]: https://github.com/home-assistant/core/pull/70598

{% enddetails %}

{% details "Energy" %}

The internal cost and compensation sensor entities, used by the energy
dashboard of Home Assistant, are now hidden by default.

([@frenck] - [#70378]) ([documentation](/integrations/energy))

[@frenck]: https://github.com/frenck
[#70378]: https://github.com/home-assistant/core/pull/70378

{% enddetails %}

{% details "Fibaro" %}

> Fibaro lights no longer supports deprecated `white_value`,

Reviewer 2: ```suggestion
Fibaro lights no longer support deprecated `white_value`,
```

{% enddetails %}

{% details "Flick Electric" %}

Units changed from `cents` to `¢/kWh`. This more accurately represents the type
of value, and makes the sensor work when used in the Energy dashboards.

If you relied on this unit of measurement format in your automations, scripts,
or have been recording it in an external time metrics database, you may need
to adapt to this change.

([@ZephireNZ] - [#69396]) ([documentation](/integrations/flick_electric))

[@ZephireNZ]: https://github.com/ZephireNZ
[#69396]: https://github.com/home-assistant/core/pull/69396

{% enddetails %}

{% details "History Stats" %}

The pretty formatted value attribute has been removed in favor of using the
new duration state class, which is formatted by the frontend.

This change significantly reduces the number of rows stored in the database.

([@bdraco] - [#70720]) ([documentation](/integrations/history_stats))

[@bdraco]: https://github.com/bdraco
[#70720]: https://github.com/home-assistant/core/pull/70720

{% enddetails %}

{% details "HomeKit" %}

The device class `carbon_dioxide` is no longer admitted as a `binary_sensor`
device class; as it is not a valid device class for a binary sensor. If you have

> overridden the device class with customize need to adjust your configuration.

Reviewer 2: ```suggestion
overridden the device class with customize you need to adjust your configuration.
```

[#69239]: https://github.com/home-assistant/core/pull/69239

{% enddetails %}

{% details "IKEA TRÅDFRI" %}

The native IKEA TRÅDFRI groups are now removed. We propose using
[light groups](/integrations/group/) instead.

Additionally, the previously deprecated YAML configuration of the IKEA TRÅDFRI
integration has been removed.

IKEA TRÅDFRI is now configured via the UI, any existing YAML
configuration has been imported in previous releases and can now be safely
removed from your YAML configuration files.

([@ggravlingen] - [#68033]) ([documentation](/integrations/tradfri))

[@ggravlingen]: https://github.com/ggravlingen
[#68033]: https://github.com/home-assistant/core/pull/68033

{% enddetails %}

{% details "Integration - Riemann sum integral" %}

The Integration sensor `unit` configuration parameter was deprecated in
Home Assistant Core release 2021.10 and is now removed as it did not
take `unit_prefix` and `unit_time` into account.

Example:

```yaml
sensor:
  - platform: integration
    source: sensor.current_power
    name: energy_spent
    unit_prefix: k
    unit_time: h
    unit: MWs
    round: 2
```

The configuration in the example would cause the source sensor's unit

> (e.g., `W`) tobe overruled and the integration sensor's `unit_of_measurement`

Reviewer 2: ```suggestion
(e.g., `W`) to be overruled and the integration sensor's `unit_of_measurement`
```

`min_max` sensors now only set attributes based on the configured type of sensor
as below:

- `min`: `min_entity_id`
- `max`: `max_entity_id`
- `last`: `last_entity_id`

The following attributes are no longer present: `min`, `max`, `mean`, `median`,
`last`, `count_sensors`

If access to the data previously provided by the attributes is needed,
create a separate `min_max` sensor for that required type instead.

([@bdraco] - [#70142]) ([documentation](/integrations/min_max))

[@bdraco]: https://github.com/bdraco
[#70142]: https://github.com/home-assistant/core/pull/70142

{% enddetails %}

{% details "MJPEG IP Camera" %}

The previously deprecated YAML configuration of the MJPEG IP Camera
integration has been removed.

MJPEG IP Camera is now configured via the UI, any existing YAML
configuration has been imported in previous releases and can now be safely
removed from your YAML configuration files.

([@frenck] - [#68033]) ([documentation](/integrations/mjpeg))

[@frenck]: https://github.com/frenck
[#68980]: https://github.com/home-assistant/core/pull/68980

{% enddetails %}

{% details "ONVIF" %}

> Additionally, the previously deprecated YAML configuration of the ONVIF

Reviewer 2: ```suggestion
The previously deprecated YAML configuration of the ONVIF
```
